static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v2.js

```diff
@@ -48,63 +50,60 @@ define([
-            topCusp = (inViewT &&
-                ((window.pageYOffset + (bonzo.viewport().height * 0.4) - adHeight) > this.$adSlot.offset().top)) ?
-                'true' : 'false';
-            bottomCusp = (inViewB &&
-                (window.pageYOffset + (bonzo.viewport().height * 0.5)) < this.$adSlot.offset().top) ?
-                'true' : 'false';
-            bottomScroll = (bottomCusp === 'true') ?
-                50 - ((window.pageYOffset + (bonzo.viewport().height * 0.5) - this.$adSlot.offset().top) * -0.2) : 50;
-            topScroll = (topCusp === 'true') ?
-                ((window.pageYOffset + (bonzo.viewport().height * 0.4) - this.$adSlot.offset().top - adHeight) * 0.2) : 0;
-        }.bind(this));
+        var scrollY = window.pageYOffset,
+            viewportHeight = bonzo.viewport().height,
+            adSlotTop = this.$adSlot.offset().top,
+            that = this;
+
+        var adHeight = (this.isClosed) ? this.closedHeight : this.openedHeight;
+        var inViewB = ((scrollY + viewportHeight) > adSlotTop);
+        var inViewT = ((scrollY - (adHeight * 2)) < adSlotTop + 20);
+        var topCusp = (inViewT &&
+            ((scrollY + (viewportHeight * 0.4) - adHeight) > adSlotTop)) ?
+            'true' : 'false';
+        var bottomCusp = (inViewB &&
+            (scrollY + (viewportHeight * 0.5)) < adSlotTop) ?
+            'true' : 'false';
+        var bottomScroll = (bottomCusp === 'true') ?
+            50 - ((scrollY + (viewportHeight * 0.5) - adSlotTop) * -0.2) : 50;
+        var topScroll = (topCusp === 'true') ?
+            ((scrollY + (viewportHeight * 0.4) - adSlotTop - adHeight) * 0.2) : 0;
 
         switch (this.params.backgroundImagePType) {
             case 'split':
+                this.scrollAmount = bottomScroll + topScroll + '%';
                 fastdom.write(function () {
-                    this.scrollAmount = bottomScroll + topScroll + '%';
                     $('.ad-exp--expand-scrolling-bg').css('background-repeat', 'no-repeat');
-                }.bind(this));
+                });
                 break;
             case 'fixed':
-                fastdom.read(function () {
-                    this.scrollAmount = (window.pageYOffset - this.$adSlot.offset().top) + 'px';
-                }.bind(this));
+                this.scrollAmount = (scrollY - adSlotTop) + 'px';
                 break;
             case 'parallax':
-                fastdom.read(function () {
-                    this.scrollAmount = ((window.pageYOffset - this.$adSlot.offset().top) * 0.15) + '%';
-                }.bind(this));
+                this.scrollAmount = ((scrollY - adSlotTop) * 0.15) + '%';
                 break;
         }
+
         fastdom.write(function () {
-            $('.ad-exp--expand-scrolling-bg').css('background-position', '50%' + this.scrollAmount);
-        }.bind(this));
+            $('.ad-exp--expand-scrolling-bg').css('background-position', '50%' + that.scrollAmount);
```

Comment:
I hate `var that = this`.
Here's a question for @uplne or @Calanthe maybe, `this.scrollAmount` seems to be used only inside this function. So instead of doing

```
switch () {
   case '': this.scrollAmount = 1;
   case '': this.scrollAmount = 2;
}
fastdom.write(() => { that.scrollAmount })
```

Can we use instead
```
var scrollAmount;
switch () {
   case '': scrollAmount = 1;
   case '': scrollAmount = 2;
}
fastdom.write(() => { scrollAmount })
```

Or am I missing something?